crates/rust-analyzer/src/reload.rs

```diff
@@ -848,6 +857,10 @@ pub(crate) fn load_proc_macro(
         ) -> Result<tt::Subtree, ProcMacroExpansionError> {
             Ok(subtree.clone())
         }
+
+        fn should_expand(&self) -> bool {
```

Comment:
> so they should never be expanded anyways

Is this certain? I focused on attribute macros and a specific use-case (ignored macros) in this PR, so I don't know where else these expanders might be used. If empty or identity expansions aren't needed anywhere, I can merge them into a `NoopExpander` or something that will always return an error (or alternaitively identity/empty tree upon agreement).

Reply:
Perhaps we could keep the `IdentityExpander` for unresolved macros and other unhappy paths only, and then keep regular macros as-is but also augment ignored ones so that `should_expand` returns false, this way we can still have the ability to expand even ignored macros on demand (e.g. via a command).

Reply:
> Is this certain? 

Well disabled means disabled, if we still call this somewhere that's more of a bug imo